### DISCUSSION TITLE
DriveRecipient is inaccessible because of protection level

### DISCUSSION THREAD

**Discussion by GaddeChaitanya** [pnp/pnpcore#909] [Q&A] [answered]
I was trying to use "CreateSharingInviteAsync" method with InviteOptions.

The issue I am running into is I cannot set the Recipients property of InviteOptions because of DriveRecipient protection level.

Appreciate any sort of advice here.

Below is the code I am using to create a sharing link.

 var driveRecipients = new List<IDriveRecipient>()
            {
                new DriveRecipient
                {
                    Email = "test@test.com"
                }
            };
            var shareRequestOptions = new InviteOptions()
            {
                Message = "I'd like to share this file with you",
                RequireSignIn = false,
                SendInvitation = false,
                Recipients = driveRecipients,
                Roles = new List<PermissionRole> { PermissionRole.Read },
                //ExpirationDateTime = DateTime.Now.AddDays(5)
            };

            var share = await folder.CreateSharingInviteAsync(shareRequestOptions);

**Comment by jansenbe** [comment, marked as answer]
@GaddeChaitanya : thanks for using PnP Core SDK and providing feedback, that indeed is a bug. I've just fixed this. See https://github.com/pnp/pnpcore/blob/dev/src/sdk/PnP.Core.Test/Security/PermissionTests.cs#L288-L299 for how this will work. Today I'll ship version 1.7, so you'll have to wait for that to use the updated version.

**Reply by jansenbe** [reply]
@GaddeChaitanya : version 1.7 just shipped

**Reply by GaddeChaitanya** [reply]
> @GaddeChaitanya : thanks for using PnP Core SDK and providing feedback, that indeed is a bug. I've just fixed this. See https://github.com/pnp/pnpcore/blob/dev/src/sdk/PnP.Core.Test/Security/PermissionTests.cs#L288-L299 for how this will work. Today I'll ship version 1.7, so you'll have to wait for that to use the updated version.

Thank you for your quick response. The shared option resolves the issue related to setting the driver recipients.
